apps/docs/docs/learn/04-styling-ui/03-recipes.mdx

```diff
@@ -0,0 +1,152 @@
+---
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+sidebar_position: 5
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Recipes
+
+This section provides practical examples and patterns that demonstrate how to use StyleX for common styling scenarios.
+
+## Contextual Styles
+
+Contextual styles are an effective way to dynamically adapt styles based on a component's state or its environment. Here are some approaches to achieve this:
+
+### Using Context
+
+Context can help reduce prop-drilling by sharing state across components. For example, you can manage the open or closed state of a sidebar using React Context and conditionally apply styles:
+
+```tsx
+import { createContext, useContext } from 'react';
+import * as stylex from '@stylexjs/stylex';
+
+const SidebarContext = createContext(false);
+
+const styles = stylex.create({
+  sidebarOpen: {
+    width: '250px',
+  },
+  sidebarClosed: {
+    width: '50px',
+  },
+});
+
+function Sidebar({ children }) {
+  const isOpen = useContext(SidebarContext);
+  return (
+    <div {...stylex.props(isOpen ? styles.sidebarOpen : styles.sidebarClosed)}>
+      {children}
+    </div>
+  );
+}
+```
+
+## Variants
+
+The "variants" pattern allows you to conditionally apply one of several predefined styles based on a value. This is especially useful for theming or dynamic component behavior.
+
+### Example: Button Variants
+
+Here’s how you can create a button component with different visual styles based on a `variant` prop:
+
+```tsx
+import * as stylex from '@stylexjs/stylex';
+
+const styles = stylex.create({
+  primary: {
+    backgroundColor: 'blue',
+    color: 'white',
+    ':hover': {
+      backgroundColor: 'darkblue',
+    },
+  },
+  secondary: {
+    backgroundColor: 'gray',
+    color: 'white',
+    ':hover': {
+      backgroundColor: 'darkgray',
+    },
+  },
+});
+
+function Button({ variant = 'primary', ...props }) {
+  return <button {...props} {...stylex.props(styles[variant])} />;
+}
+
+// Usage
+<Button variant="primary">Primary</Button>
+<Button variant="secondary">Secondary</Button>
+```
+
+## Hover-Dependent Descendant Styles
+
+Using CSS variables, you can style descendants based on a parent's state, such as `:hover`.
+
+### Example: Nested Hover Styles
+
+In this example, we'll apply a hover effect to a child element when the parent is hovered:
+
+```tsx
+import * as stylex from '@stylexjs/stylex';
+
+const styles = stylex.create({
+  parent: {
+    '--child-color': 'black',
+    ':hover': {
+      '--child-color': 'blue',
+    },
+  },
+  child: {
+    color: 'var(--child-color)',
+  },
+});
```

Comment:
Change this example to use a separate `.stylex.ts` file that defines a variable using `stylex.defineVars`. We discourage the use of arbitrary strings for variables.